airbyte-webapp/src/packages/cloud/services/auth/AuthService.tsx

```diff
@@ -122,47 +123,62 @@ export const AuthenticationProvider: React.FC = ({ children }) => {
   const auth = useAuth();
   const userService = useGetUserService();
   const analytics = useAnalyticsService();
-  const authService = useMemo(() => new GoogleAuthService(() => auth), [auth]);
+  const authServiceRef = useRef(new GoogleAuthService(() => auth));
```

Comment:
This is currently constructing a new service instance on every render of this provider, i.e. roughly 6 times on each load. Unfortunately `useRef` does not support a callback for init (and `react-use` doesn't have a hook for that either). I'd suggest we add a `useSingleton` hook (like https://github.com/elastic/kibana/blob/main/src/plugins/discover/public/application/main/utils/use_singleton.ts) to our utils and use this here, so we only need to call the GoogleAuthService constructor once.